cmd/query/app/grpc_handler.go

```diff
@@ -57,6 +57,39 @@ type GRPCHandler struct {
 	nowFn               func() time.Time
 }
 
+// GRPCHandlerOptions contains optional members of GRPCHandler.
+type GRPCHandlerOptions struct {
+	Logger *zap.Logger
+	Tracer opentracing.Tracer
+	NowFn  func() time.Time
+}
+
+// NewGRPCHandler returns a GRPCHandler.
+func NewGRPCHandler(queryService *querysvc.QueryService,
+	metricsQueryService querysvc.MetricsQueryService,
+	options GRPCHandlerOptions,
+) *GRPCHandler {
+	if options.Logger == nil {
+		options.Logger = zap.NewNop()
+	}
+
+	if options.Tracer == nil {
+		options.Tracer = opentracing.NoopTracer{}
+	}
+
```

Comment:
please add a test that passes empty options struct and validates that all 3 fields are not-nil in the result